src/pages/ProductInner/ProductInner.js

```diff
@@ -0,0 +1,88 @@
+import React, { useState, useEffect } from 'react';
+import { useParams } from 'react-router-dom';
+import { Link } from 'react-router-dom';
+
+import { Preloader } from '../../components/Preloader/Preloader';
+import { ErrorMessage } from '../../components/ErrorMesage/ErrorMessage';
+
+import './ProductInner.scss';
+
+export function ProductInner() {
+  const { id } = useParams();
+  const [product, setPdouct] = useState(null);
+  const [productImageUrl, setPdouctImageUrl] = useState('');
+  const [loading, setLoading] = useState(true);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    fetch(`https://api.thedogapi.com/v1/breeds/${id}`)
+      .then((res) => {
+        setLoading(false);
+        if (!res.ok) throw new Error();
+        return res.json();
+      })
+      .then((data) => setPdouct(data))
+      .catch(() => setIsError(true));
+  }, [id]); // why id?
+
+  useEffect(() => {
+    product &&
+      fetch(`https://api.thedogapi.com/v1/images/${product.reference_image_id}`)
+        .then((res) => {
+          if (!res.ok) throw new Error();
+          return res.json();
+        })
+        .then((data) => setPdouctImageUrl(data.url))
+        .catch(() => setIsError(true));
```

Comment:
@dima0208 I agree with Artur, you could define some function similar to getPosts that you already have and use it here

for example, getProductDetails and getProductImage that will return the result you will use in this ProductInner component

or least you can create a common api helper that will return the result of the fetch operation, throw an error in this helper and return res.json() to prevent the duplication of the code